### PR TITLE
Fix the issue with linking against both Lightning-Kokkos and Lightning 

### PR DESCRIPTION
**Pull request by maliasadi**
Currently, we cannot build and link both Lightning and Lightning-Kokkos in a single translation unit due to conflicts in the name of files, functions, classes, and namespaces. This PR addresses this issue. A direct use case of this patch would be in getting support for both `lightning.qubit` and `lightning.kokkos` simulators in Catalyst.

[sc-37004]

### CODE REVIEW

File: pennylane_lightning_kokkos/src/bindings/Bindings.cpp [hunk 1]
```diff
@@ -65,21 +65,21 @@ void StateVectorKokkos_class_bindings(py::module &m) {
...
         .def(py::init([](std::size_t num_qubits,
-                         const Kokkos::InitArguments &kokkos_args) {
+                         const ::Kokkos::InitArguments &kokkos_args) {
             return new StateVectorKokkos<PrecisionT>(num_qubits, kokkos_args);
         }))
         .def(py::init([](const np_arr_c &arr) {
             py::buffer_info numpyArrayInfo = arr.request();
-            auto *data_ptr =
-                static_cast<Kokkos::complex<PrecisionT> *>(numpyArrayInfo.ptr);
+            auto *data_ptr = static_cast<::Kokkos::complex<PrecisionT> *>(
```
Comment (vincentmr) [inline pennylane_lightning_kokkos/src/bindings/Bindings.cpp:73]: Should we define somewhere
```c++
using CFP_t = ::Kokkos::complex<PrecisionT>;
```
and just reuse that?

File: pennylane_lightning_kokkos/src/simulator/GateFunctors.hpp [hunk 2]
```diff
@@ -3,29 +3,29 @@
...
 #include <Kokkos_StdAlgorithms.hpp>
 
-#include "Util.hpp"
+#include "UtilKokkos.hpp"
 
 namespace {
-using namespace Pennylane::Util;
-namespace KE = Kokkos::Experimental;
+using namespace Pennylane::Lightning::Kokkos::Util;
+namespace KE = ::Kokkos::Experimental;
```
Comment (vincentmr) [inline pennylane_lightning_kokkos/src/simulator/GateFunctors.hpp:10]: Why do we need global scope access via operator:: everywhere (including here)?

File: pennylane_lightning_kokkos/src/simulator/MeasuresFunctors.hpp [hunk 3]
```diff
@@ -58,25 +58,25 @@ template <class Precision> struct getCDFFunctor {
 /**
  *@brief Sampling using Random_XorShift64_Pool
  *
- * @param samples_ Kokkos::View of the generated samples.
- * @param cdf_  Kokkos::View of cumulative probability distribution.
+ * @param samples_ ::Kokkos::View of the generated samples.
```
Comment (vincentmr) [inline pennylane_lightning_kokkos/src/simulator/MeasuresFunctors.hpp:61]: Do we want to have operator:: in the docstrings?

**Review by vincentmr — APPROVED** [review]
Looks good to me. I have a few questions, essentially about formatting. I'm guessing that `Kokkos::` was replaced by `::Kokkos::` with `sed` or a similar command without much discrimination. I'm wondering if any issue could arise from that in the long run, or if it shows poor style (e.g. in the docstrings). I'd be OK merging as is, but I wonder what are people's thoughts.

File: pennylane_lightning_kokkos/src/bindings/Bindings.cpp (hunk 1 above)
Comment (mlxd) [reply pennylane_lightning_kokkos/src/bindings/Bindings.cpp:73]: Agreed. That should probably be the best option, assuming it doesn't interfere elsewhere.

File: pennylane_lightning_kokkos/src/algorithms/AdjointDiffKokkos.cpp [hunk 4]
```diff
@@ -1,5 +1,7 @@
 #include "AdjointDiffKokkos.hpp"
 
 // explicit instantiation
-template class Pennylane::Algorithms::AdjointJacobianKokkos<float>;
-template class Pennylane::Algorithms::AdjointJacobianKokkos<double>;
+template class Pennylane::Lightning::Kokkos::Algorithms::AdjointJacobianKokkos<
```
[the remark this answers is not in the document]
Comment (maliasadi) [reply pennylane_lightning_kokkos/src/algorithms/AdjointDiffKokkos.cpp:4]: I see your points @AmintorDusko.
I've chatted with @mlxd regarding these namespaces a while ago and we initially thought that using `::Kokkos` wouldn't add any complexity. For lightning repositories, which of these two options seem better? 
Option 1.  `Lightning`, `Lightning::GPU`, and `Lightning::Kokkos`  (conflicting with kokkos namespace) or 
Option 2. `Lightning`, `Lightning_GPU`, and `Lightning_Kokkos`  
@mlxd What are your thoughts?

File: pennylane_lightning_kokkos/src/bindings/Bindings.cpp (hunk 1 above)
Comment (maliasadi) [reply pennylane_lightning_kokkos/src/bindings/Bindings.cpp:73]: Nice suggestion! Fixing this, Thanks @vincentmr!

File: pennylane_lightning_kokkos/src/simulator/MeasuresFunctors.hpp (hunk 3 above)
Comment (maliasadi) [reply pennylane_lightning_kokkos/src/simulator/MeasuresFunctors.hpp:61]: Nah! Fixing this too.

File: pennylane_lightning_kokkos/src/algorithms/AdjointDiffKokkos.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (vincentmr) [reply pennylane_lightning_kokkos/src/algorithms/AdjointDiffKokkos.cpp:4]: I think I'll vote for option #2.

[the remark this answers is not in the document]
Comment (mlxd) [reply pennylane_lightning_kokkos/src/algorithms/AdjointDiffKokkos.cpp:4]: Given the comments on `::Kokkos`, if the team will be maintaining this, let's go for the consensus here of Option 2.

[the remark this answers is not in the document]
Comment (maliasadi) [reply pennylane_lightning_kokkos/src/algorithms/AdjointDiffKokkos.cpp:4]: Thanks everyone! :+1:

**Comment by maliasadi** [comment]
**Note** 
Those broken Python tests aren't related to the changes in this PR and they'll be fixed in PR #60.

**Review by multiphaseCFD — APPROVED** [review]
Nice work @maliasadi! Thanks for putting this together!

**Pull request merged**